src/frontend/screens/Settings/components/OtherSettings/UseDGPU.tsx

```diff
@@ -6,22 +6,41 @@ import ContextProvider from 'frontend/state/ContextProvider'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { faCircleInfo } from '@fortawesome/free-solid-svg-icons'
 
-const PrimeRun = () => {
+const UseDGPU = () => {
   const { t } = useTranslation()
-  const { platform } = useContext(ContextProvider)
+  const { platform, showDialogModal } = useContext(ContextProvider)
   const isLinux = platform === 'linux'
-  const [primeRun, setPrimeRun] = useSetting<boolean>('primeRun', false)
+
+  const [useDGPU, setUseDGPU] = useSetting('nvidiaPrime', false)
 
   if (!isLinux) {
     return <></>
   }
 
+  async function toggleUseDGPU() {
+    const numOfGpus = await window.api.getNumOfGpus()
+    if (!useDGPU && numOfGpus === 1) {
+      showDialogModal({
+        title: t('setting.primerun.confirmation.title', 'Only 1 GPU detected'),
+        message: t(
+          'setting.primerun.confirmation.message',
+          'Only one graphics card was detected in this system. ' +
+            'Please note that this option is intended for multi-GPU systems with headless GPUs (like laptops). ' +
+            'On single-GPU systems, the GPU is automatically used & enabling this option can cause issues'
+        ),
+        buttons: [{ text: t('box.ok', 'Ok') }],
```

Comment:
Maybe we can change this `Ok` button to have to buttons: Confirm and Cancel, and only set the state if `Confirm` is clicked?

I think it's more user friendly to warn the user and make them confirm the change instead of applying the change and if they realize it's a bad idea then they have to uncheck it. User's may not realize that it actually changed it it was some mistake or other checkboxes are checked around it.